example/src/main/java/org/xmtp/android/example/conversation/ConversationDetailViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package org.xmtp.android.example.conversation
+
+import androidx.annotation.UiThread
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import org.xmtp.android.library.Conversation
+
+class ConversationDetailViewModel : ViewModel() {
+    var conversation: Conversation? = null
+
+    private val _uiState = MutableStateFlow<UiState>(UiState.Loading(null))
+    val uiState: StateFlow<UiState> = _uiState
+
+    @UiThread
+    fun fetchMessages(peerAddress: String) {
+        when (val uiState = uiState.value) {
+            is UiState.Success -> _uiState.value = UiState.Loading(uiState.listItems)
+            else -> _uiState.value = UiState.Loading(null)
+        }
+        viewModelScope.launch(Dispatchers.IO) {
+            val listItems = mutableListOf<MessageListItem>()
+            try {
+                conversation?.let {
+                    it.messages().map { message ->
```

Comment:
@nplasterer i'm wondering if maybe we can get a single conversation from a client here instead of having to pass a convo from activity to activity 💭 . Either that or I think we'll want a way to make Conversation parcelizable so we can pass it in and call `messages()` on it directly. Will be fun to pair on this together!